src/journal/journald-stream.c

```diff
@@ -511,8 +511,8 @@ static int stdout_stream_process(sd_event_source *es, int fd, uint32_t revents,
                 goto terminate;
         }
 
-        /* If the buffer is full already (discounting the extra NUL we need), add room for another 1K */
-        if (s->length + 1 >= s->allocated) {
+        /* If the buffer is almost full, add room for another 1K */
+        if (s->length + 1024 > s->allocated) {
```

Comment:
this doesn't look right. let's say we get each byte piecemeal, i.e. one by one. Previously we'd allocate 1025 byte, then one by one fill it up to 1024. Then we'd re-allocate to 2049 and fill it to one by one to 2048 bytes, and then re-allocate again to 3073, and so on. 

With your changed code we'd start with 1025, read a single byte, then immediately reallocate to 1026, and read one more byte, reallocate to 1027, and so on.

net result is we'd do n times as much work for n bytes. That's not what you want no?

I figure what you actually want here is when there are less than 1K free allocate 2K at once.

Reply:
Actually the first allocation is equal to `((1024 + 1) * 2 + Extra)`
My only goal is to have at least ~1KB a free available space in the buffer to append data (Put another way, I want that the read buffer have a minimal length of ~1KB).
I do not want to have a buffer of 3 bytes, which is not really efficient.